### PR TITLE
Change reccurrent adding and duplication algoritm. Fixed Cancellation

### PR DESCRIPTION
**Pull request by hrybs**
LVJAVS-684 - http://ssu-jira.softserveinc.com/browse/LVJAVS-684
LVJAVS-695 - http://ssu-jira.softserveinc.com/browse/LVJAVS-695
LVJAVS-707 - http://ssu-jira.softserveinc.com/browse/LVJAVS-707

### CODE REVIEW

File: src/main/java/ua/softserveinc/tc/dto/BookingDto.java [hunk 1]
```diff
@@ -54,51 +52,58 @@ public BookingDto() {
...
-        this.idChild = booking.getChild().getId();
-        this.comment = booking.getComment();
-        this.recurrentId = booking.getRecurrentId();
+        startTime = df.format(booking.getBookingStartTime());
+        endTime = df.format(booking.getBookingEndTime());
+        startTimeMillis = booking.getBookingStartTime().getTime();
+        endTimeMillis = booking.getBookingEndTime().getTime();
+        kidName = booking.getChild().getFullName();
+        roomName = booking.getRoom().getAddress();
+        durationBooking = endTimeMillis - startTimeMillis;
```
Comment (andryspo) [inline src/main/java/ua/softserveinc/tc/dto/BookingDto.java]: check if null

File: src/main/webapp/WEB-INF/views/errors/common-message.jsp [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<%@ taglib uri="http://www.springframework.org/tags" prefix="spring"%>
+
+<div  class="message">
+    <img class="not-found" src="resources/img/sorry.png" />
+    <br>
+    <spring:message code="error.common" />
+</div>
```
Comment (andryspo) [inline src/main/webapp/WEB-INF/views/errors/common-message.jsp]: empty line

File: src/main/java/ua/softserveinc/tc/dto/BookingDto.java (hunk 1 above)
Comment (hrybs) [inline src/main/java/ua/softserveinc/tc/dto/BookingDto.java]: Thanks for advise.

File: src/main/webapp/WEB-INF/views/errors/common-message.jsp (hunk 2 above)
Comment (hrybs) [inline src/main/webapp/WEB-INF/views/errors/common-message.jsp]: Thanks for advise.

**Pull request merged**